### PR TITLE
Fix language preferences (langfix)

### PR DESCRIPTION
**Pull request by timbl**
Fix language preferences so that they always include the defaults after the use's choice.

Add a few tests

Responding to https://github.com/solid/solidos/issues/42

Needs more tests ...

### CODE REVIEW

File: test/unit/widgets/forms/autocomplete/language.test.ts [hunk 1]
```diff
@@ -61,24 +63,37 @@ const bobProfileText = prefixes + `
...
   it('exists as a function', () => {
     expect(getPreferredLanagugesFor).toBeInstanceOf(Function)
   })
   it('returns just greek for Alice, plus deafults', async () => {
     const result = await getPreferredLanagugesFor(alice)
-    expect(result).toEqual(['el'].concat(['en', 'fr', 'de', 'it', 'ar'])) // toMatchSnapshot()
+    expect(result).toEqual(['el'].concat(['en', 'fr', 'de', 'it', 'ar']))
   })
-  it('returns just greek for Bob, plus deafults', async () => {
+  it('returns english, french Bob, plus deafults', async () => {
```
Comment (TallTed) [inline test/unit/widgets/forms/autocomplete/language.test.ts:120]:
```suggestion
  it('returns english, french Bob, plus defaults', async () => {
```

File: test/unit/widgets/forms/autocomplete/language.test.ts [hunk 2]
```diff
@@ -61,24 +63,37 @@ const bobProfileText = prefixes + `
...
+  it('exists as a array', () => {
+    expect(defaultPreferedLangages).toEqual(expectedDefaults)
+  })
+})
+
 describe('getPreferredLanagugesFor', () => {
   it('exists as a function', () => {
     expect(getPreferredLanagugesFor).toBeInstanceOf(Function)
   })
   it('returns just greek for Alice, plus deafults', async () => {
```
Comment (TallTed) [inline test/unit/widgets/forms/autocomplete/language.test.ts]:
```suggestion
  it('returns just greek for Alice, plus defaults', async () => {
```

File: test/unit/widgets/forms/autocomplete/language.test.ts [hunk 3]
```diff
@@ -61,24 +63,37 @@ const bobProfileText = prefixes + `
...
-    expect(result).toEqual(['el'].concat(['en', 'fr', 'de', 'it', 'ar'])) // toMatchSnapshot()
+    expect(result).toEqual(['el'].concat(['en', 'fr', 'de', 'it', 'ar']))
   })
-  it('returns just greek for Bob, plus deafults', async () => {
+  it('returns english, french Bob, plus deafults', async () => {
     const result = await getPreferredLanagugesFor(bob)
-    expect(result).toEqual(['en', 'fr', 'de', 'it', 'ar']) // toMatchSnapshot()
+    expect(result).toEqual(['en', 'fr', 'de', 'it', 'ar'])
+  })
+  it('returns nothing for Charlie, plus deafults', async () => {
```
Comment (TallTed) [inline test/unit/widgets/forms/autocomplete/language.test.ts:124]:
```suggestion
  it('returns nothing for Charlie, plus defaults', async () => {
```

File: src/widgets/forms/autocomplete/language.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ export async function getPreferredLanagugesFor (person: NamedNode) {
   const list = kb.any(person, ns.schema('knowsLanguage'), null, doc) as Collection | undefined
   if (!list) {
     console.log(`User ${person} has not set their languages in their profile.`)
-    return null // differnet from []
+    return defaultPreferedLangages
```
Comment (TallTed) [inline src/widgets/forms/autocomplete/language.ts:40]: Creative spellings --
* `defaultPreferedLangages` 
* `getPreferredLanagugesFor`

There may be others.  To fix the above ---

* `s/Prefered/Preferred/g`
* `s/Langages/Languages/g`
* `s/Lanaguges/Languages/g`

Comment (timbl) [reply src/widgets/forms/autocomplete/language.ts:40]: Opps. global change made. Thanks!

**Comment by timbl** [comment]
I'll merge this before getting into the auth upgrade.

**Comment by bourgeoa** [comment]
Ok

**Pull request merged**